### DISCUSSION TITLE
Override local/global search prompts

### DISCUSSION THREAD

**Discussion by MisterAndry** [microsoft/graphrag#1146] [Q&A]
Hi everyone!
I'm curious if there is a way to override local/global system prompts?
If it is possible, how?
The option of changing the grafrag source code doesn't work for me, as my application loads all libraries and dependencies when it is deployed to the server.

**Comment by ganeshrohan** [comment]
Yes, there is an option to provide a custom system prompt in local search - [https://github.com/microsoft/graphrag/blob/737a471d18849eb68620217d8f6b6be1972131c0/graphrag/query/structured_search/local_search/search.py#L40](url) 

```
search_engine = LocalSearch(
    llm=llm,
    context_builder=context_builder,
    token_encoder=token_encoder,
    llm_params=llm_params,
    context_builder_params=local_context_params,
    response_type="prioritized list"  # free form text describing the response type and format, can be anything, e.g. prioritized list, single paragraph, multiple paragraphs, multiple-page report
    system_prompt=graphrag_system_propmpt_instruct
)
```
